src/component-store/lifecycle-hooks.ts

```diff
@@ -0,0 +1,12 @@
+import { ComponentStore, provideComponentStore } from '@ngrx/component-store';
```

Comment:
Hey, thanks for creating this PR.
Could you rename this file to `provide-immer-component-store.ts` please.
The reason for this is that it isn't a lifecycle hook (it seems like it was added within the same file in the NgRx repo 🤔).

I would also ask to move the class instance to a new file `immer-component-store.ts`.
The reason for this is that I'm not a fan that both files have an import to the other file.
The `index` barrel file can then export both stores.
What do you think?